stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -463,6 +465,24 @@ private void handleDateValue(int index, BStruct bStruct, RegistryIndex stringReg
         }
     }
 
+    private void handleBinaryValue(BStruct bStruct, RegistryIndex refRegIndex, RegistryIndex blobRegIndex, int index)
+            throws SQLException {
+        BStructType.StructField[] structFields = getStructFields();
+        int fieldTypeTag = getFieldTypeTag(structFields, index);
+        byte[] binaryValue = rs.getBytes(index);
+        if (fieldTypeTag == TypeTags.UNION_TAG) {
+            BRefType refValue = binaryValue == null ? null : new BBlob(binaryValue);
+            validateAndSetRefRecordField(bStruct, refRegIndex.incrementAndGet(), TypeTags.BLOB_TAG,
+                    retrieveNonNilTypeTag(structFields, index - 1), refValue, UNASSIGNABLE_UNIONTYPE_EXCEPTION);
+        } else {
+            if (binaryValue != null) {
+                bStruct.setBlobField(blobRegIndex.incrementAndGet(), binaryValue);
+            } else {
+                handleNilToNonNillableFieldAssignment();
+            }
+        }
+    }
+
```

Comment:
Instead of having two methods, can we have a single method which includes this logic which accepts a byte[] as a parameter?